qualang_tools/loops/loops.py

```diff
@@ -0,0 +1,205 @@
+    else:
+        return var, start, var > stop_condition, var + step
+
+
+def qua_linspace(var, start, stop, num):
+    """Function parametrizing the QUA `for_` loop from the numpy.linspace() syntax.
+
+    :param var: the QUA variable that will be looped over (int or fixed).
+    :param start: The starting value of the sequence. The interval includes this value. Must be a Python variable.
+    :param stop: The end value of the sequence. The interval includes this value within the 2**-28 fixed point accuracy. Must be a Python variable.
+    :param num: Number of samples to generate. Must be a Python variable.
+    :return: QUA for_ loop parameters (var, init, cond, update) as defined in https://qm-docs.qualang.io/api_references/qua/dsl_main?highlight=for_#qm.qua._dsl.for_.
+    """
+    # Check QUA vs python variables
+    if not hasattr(var, "_exp"):
+        raise Exception("The first argument must be a QUA variable.")
+    if hasattr(start, "_exp") or hasattr(stop, "_exp") or hasattr(num, "_exp"):
+        raise Exception("The for loop arguments must be python variables.")
+    # Check for fixed number overflows
+    if not (-8 <= start < 8) and not (-8 <= stop < 8):
+        raise Exception("fixed numbers are bounded to [-8, 8).")
+    # Check if num is integer
+    if not float(num).is_integer():
+        raise Exception("The number of samples must be a python integer.")
+
+    if num > 1:
+        step = (stop - start) / (num - 1)
+    elif num == 1:
+        step = (stop - start) * 2
+    else:
+        step = stop - start
+
+    if step > 0:
+        return var, start, var < stop + step / 2, var + step
+    else:
+        return var, start, var > stop + step / 2, var + step
+
+
+def qua_logspace(var, start, stop, num):
+    """Function parametrizing the QUA `for_` loop from the numpy.logspace() syntax.
+
+    :param var: the QUA variable that will be looped over (fixed).
+    :param start: The starting value of the sequence. The interval includes this value. Must be a Python variable.
+    :param stop: The end value of the sequence. The interval includes this value within the 2**-28 fixed point accuracy. Must be a Python variable.
+    :param num: Number of samples to generate. Must be a Python variable.
+    :return: QUA for_ loop parameters (var, init, cond, update) as defined in https://qm-docs.qualang.io/api_references/qua/dsl_main?highlight=for_#qm.qua._dsl.for_.
+    """
+    # Check QUA vs python variables
+    if not hasattr(var, "_exp"):
+        raise Exception("The first argument must be a QUA variable.")
+    if hasattr(start, "_exp") or hasattr(stop, "_exp") or hasattr(num, "_exp"):
+        raise Exception("The for loop arguments must be python variables.")
+    # Check for fixed number overflows
+    if not (-8 <= start < 8) and not (-8 <= stop < 8):
+        raise Exception("fixed numbers are bounded to [-8, 8).")
+    # Check if num is integer
+    if not float(num).is_integer():

```

Comment:
I wondered if his would work if someone put a QUA variable, and it would fail.
Trying to do "float" on a QUA variable returns an error, so instead of the nice error you wrote, they would get a generic one...
All of these need to change into `isinstance` as I wrote above